unstructured/partition/utils/ocr_models/ocr_interface.py

```diff
@@ -1,44 +1,41 @@
+from __future__ import annotations
+
 import functools
 import importlib
 from abc import ABC, abstractmethod
-from typing import TYPE_CHECKING, Any, List
+from typing import TYPE_CHECKING
 
 from unstructured.partition.utils.constants import OCR_AGENT_MODULES_WHITELIST
 
 if TYPE_CHECKING:
-    from PIL import PILImage
+    from PIL import Image as PILImage
     from unstructured_inference.inference.elements import TextRegion
     from unstructured_inference.inference.layoutelement import (
         LayoutElement,
     )
 
 
 class OCRAgent(ABC):
-    def __init__(self):
-        self.agent = self.load_agent()
-
-    @abstractmethod
-    def load_agent(self, language: str) -> Any:
-        pass
```

Comment:
Why was the __init__ method (and the related abstractmethod) removed?

Reply:
In a nutshell, because this class specifies the interface for an "OCRAgent" and neither of those methods is actually part of the interface.

Some background facts:
- The only call to `.load_agent()` is from the initializer (`.__init__()`).
- `.load_agent() is implemented in the Paddle OCR-agent.
- `.load_agent()` is _not_ implemented in the Tessaract OCR-agent.

From those three facts we can infer:
- `.load_agent()` is not part of the interface (because you can be an agent without implementing that method).
- `.load_agent()` is part of the implementation (not interface) because it is only called internally.

For this reason, I removed these from the `OCRAgent` ABC (interface definition) and moved them to the `Paddle` implementation.

My engineering motivations were:
- An interface should be as small as possible while remaining coherent and useful.
    - If I'm implementing an interface I want to know the minimum methods I need to implement.
    - If I'm using an interface I want to know what is guaranteed to be available to me.
- An interface specifies:
    - what can be called
    - what it is called with
    - what it returns

    Everything else should be left to the discretion of the implementor.

In fairness, an ABC can also act as a base class, and `.get_instance()` is an example of an ABC providing a common method, so these principles must be applied with some judgement. But in this case, it was clear `.load_agent()` was an implementation detail so I moved it to the implementation that used it.